exist-core/src/test/xquery/xquery3/serialize.xql

```diff
@@ -234,7 +234,7 @@ function ser:serialize-empty-sequence() {
 
 declare
     %test:args(1234)
-    %test:assertEquals('"1234"')
+    %test:assertEquals(1234)
```

Comment:
Adaptive serialisation will always output a string. The test should reflect that.

Reply:
This is not related to adaptive serialization.

This is related to the XQSuite use of XQuery annotations. The XQuery 3.1 spec says that an `Annotation` can have zero or more [Literal](https://www.w3.org/TR/xquery-31/#prod-xquery31-Literal) values. Further, it says that a `Literal` is either a `StringLiteral`, `NumericLiteral`, `IntegerLiteral`, `DecimalLiteral` or `DoubleLiteral`.

If you `assertEquals(1234)` and your test returns an integer, then the integers will now be correctly compared. There was some support for this before in XQSuite but it was broken/incomplete.

Reply:
Sorry but I have to disagree, @adamretter. This test is for adaptive serialization, if it does _not_ return a string than that is a problem. A reason to reject this PR. So please change it to a string so that we can see that it does in fact return a string and compares them correctly.

Reply:
@line-o It may itself be a test for adaptive serialization, but after that the assert-equals comes into play

Reply:
Which should ensure that the expected value _and_ type is returned.